sdks/python/apache_beam/io/filesystem.py

```diff
@@ -209,7 +209,7 @@ def _fetch_to_internal_buffer(self, num_bytes):
         # objects for the unused compressed data.
         if (self._compression_type == CompressionTypes.BZIP2 or
             self._compression_type == CompressionTypes.GZIP):
-          if self._decompressor.unused_data != '':
+          if self._decompressor.unused_data != b'':
```

Comment:
This change fixes the hanging tests.

This code block resulted in an infinite loop when `self._decompressor.unused_data = b''`. On Python 2, this was not an issue since `b'' == ''`

Reply:
Thanks for catching!